src/app/content/highlights/components/Highlights.tsx

```diff
@@ -90,6 +98,62 @@ interface SectionHighlightsProps {
 
 // tslint:disable-next-line: variable-name
 export const SectionHighlights = ({ location, highlights }: SectionHighlightsProps) => {
+  const [highlightIdToEdit, setHighlightIdToEdit] = React.useState<string | null>(null);
```

Comment:
lets move all this logic down into a per-highlight component. the only reason i can think of to have the logic in the list is to do some special handling for editing or blocking the editing of multiple highlights at once, but suhani has said we don't need to do that